src/util.rs

```diff
@@ -118,6 +119,26 @@ impl MaxSize for Hash {
     const POSTCARD_MAX_SIZE: usize = 32;
 }
 
+/// A serializable error type for use in RPC responses.
+#[derive(Serialize, Deserialize, Debug)]
+pub struct RpcError(serde_error::Error);
+
+impl std::error::Error for RpcError {}
```

Comment:
Pretty sure we already depend on `thiserror`, so that seems a little cleaner to use